MovieApp/CastCollectionView.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  CastCollectionView.swift
+//  MovieApp
+//
+//  Created by Five on 06.05.2022..
+//
+
+import Foundation
+import SnapKit
+import UIKit
+import MovieAppData
+
+class CastCollectionView: UIView {
+    let cellIdentifier = "cellId"
+    let collectionView : UICollectionView = {
+        let flowlayout = UICollectionViewFlowLayout()
+        flowlayout.scrollDirection = .vertical
+        
+        return UICollectionView(frame: CGRect.zero, collectionViewLayout: flowlayout)
+    }()
+    weak var appViewCollectionViewCommunicationDelegate : appViewCastCollectionCommunication?
+    var castList : [[String]] = []
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+    }
+
+    convenience init() {
+        self.init(frame: CGRect.zero)
+        buildView()
+        setLayout()
+    }
+
+    required init(coder aDecoder: NSCoder) {
+        fatalError("This class does not support NSCoding")
+    }
+    
+    
+    func buildView() {
+        addSubview(collectionView)
+        backgroundColor = .white
+        
+        collectionView.isScrollEnabled = false
+        collectionView.register(CastView.self, forCellWithReuseIdentifier: cellIdentifier)
+        collectionView.dataSource = self
+        collectionView.delegate = self
+        collectionView.backgroundColor = .white
+    }
+    
+    
+    func setLayout() {
+        
+        collectionView.snp.makeConstraints { (make) in
+            make.top.leading.trailing.bottom.equalToSuperview()
+        }
+
+    }
+
+}
+
+
+extension CastCollectionView: UICollectionViewDataSource {
```

Comment:
I don't see a reason for this view to be a collection view anyway. It could have been a vertical stack view with view inside depending on the number of casts.